processor/tailsamplingprocessor/processor.go

```diff
@@ -484,12 +485,16 @@ func (tsp *tailSamplingSpanProcessor) iter(tickChan <-chan time.Time, workChan <
 // dropped) and forwards the span appropriately. It returns true if the trace
 // was cached and false if regular processing must be done.
 func (tsp *tailSamplingSpanProcessor) processCachedTrace(traceID pcommon.TraceID, rss ptrace.ResourceSpans, spanCount int64) bool {
-	if _, ok := tsp.sampledIDCache.Get(traceID); ok {
+	if metadata, ok := tsp.sampledIDCache.Get(traceID); ok {
 		tsp.logger.Debug("Trace ID is in the sampled cache", zap.Stringer("id", traceID))
 		traceTd := ptrace.NewTraces()
 		appendToTraces(traceTd, rss)
 		if tsp.recordPolicy {
-			sampling.SetBoolAttrOnScopeSpans(traceTd, "tailsampling.cached_decision", true)
+			if metadata.PolicyName != "" {
+				sampling.SetAttrOnScopeSpans(traceTd, "tailsampling.policy", metadata.PolicyName)
+			} else {
+				sampling.SetBoolAttrOnScopeSpans(traceTd, "tailsampling.cached_decision", true)
```

Comment:
What would you think of adding this even if a policy name is present to make it clear which spans showed up late?

Reply:
I like this idea.